spk-config.yaml

```diff
@@ -5,7 +5,26 @@ azure_devops: # Should this live under the `spk` configuration?
   manifest_repository: "dev.azure.com/bhnook/fabrikam/_git/materialized" # Repository URL that is configured for flux. This holds the kubernetes manifests that is generated by fabrikate.
   org: "https://dev.azure.com/bhnook" # Your AzDo Org
   project: "fabrikam" # Your AzDo project
-
+  variable_group:
+    name: "myvg"
+    description: "myvg description"
+    variables: # Either this or Key Value provider should be configured for this variable group
+      var1:
+        value: value1
+        isSecret: true
+      var2:
+        value: value2
+        isSecret: false
+    key_vault_provider:
+      name: "mykeyvault" # name of the Azure Key Vaukt with Secrets
+      secrets: ["secret1", "secret2"] # Names of secrets from Azure Key Vault to map in Variable Group
+      service_endpoint: # service endpoint is required to authorize with Azure Key Vault
+        name: "Bedrock-GitOps-KeyVault" # if the service endpoint with this name does not exist, the following values are required to create a new service connection with this name
+        subscription_id: "subscription-id" # Azure Subscription id where Key Vault exist
+        subscription_name: "subscription-name" # Azure Subscription name where Key Vault exist
+        service_principal_id: "service-principal-id" # Service Principal Id that has 'Get' and 'List' in Key Vault Access Policy
+        service_principal_secret: "service-principal-secret" # Service Principal secret for the above Service Principal Id
+        tenant_id: "tenant-id" # AAD Tenant Id for the above Service Principal
```

Comment:
if the name does not exist, I'm thinking all these values would be needed either as a one time, or we could reuse them from under `introspection.azure` in config.

Reply:
The Service Principal for authorizing with Key Vault can be different with minimum `get` and `list` permissions in Key Vault access policy instead of `owner` permissions requirement in `introspection` area.